corehq/apps/sso/models.py

```diff
@@ -404,6 +410,56 @@ def get_required_identity_provider(cls, username):
             return idp
         return None
 
+    def get_all_members_of_the_idp(self):
+        authority_base_url = "https://login.microsoftonline.com/"
+        authority = f"{authority_base_url}{self.api_host}"
+
+        config = {
+            "authority": authority,
+            "client_id": self.api_id,
+            "scope": ["https://graph.microsoft.com/.default"],
+            "secret": self.api_secret,
+            "endpoint": "https://graph.microsoft.com/v1.0/users"
+        }
+
+        try:
+            # Create a preferably long-lived app instance which maintains a token cache.
+            app = msal.ConfidentialClientApplication(
+                config["client_id"], authority=config["authority"],
+                client_credential=config["secret"],
+            )
+            # looks up a token from cache
+            result = app.acquire_token_silent(config["scope"], account=None)
+            if not result:
+                result = app.acquire_token_for_client(scopes=config["scope"])
+            if "access_token" in result:
+                # Calling graph using the access token
+                response = requests.get(
+                    config["endpoint"],
+                    headers={'Authorization': 'Bearer ' + result['access_token']},
+                )
+                response.raise_for_status()  # Raises an error for bad status
+                graph_data = response.json()
+
+                user_principal_names = [user["userPrincipalName"] for user in graph_data["value"]]
+                return user_principal_names
+            else:
+                error_message = f"Error acquiring token: {result.get('error')}, Description: "
+                f"{result.get('error_description')}, Correlation ID: {result.get('correlation_id')}"
+                raise Exception(error_message)
+        except Exception as e:
+            notify_exception(None, f"Failed to get members of the IDP. {str(e)}")
```

Comment:
nit: Generally `Identity Provider` is `IdP` in the reference docs elsewhere, rather than with a capital `D`
```suggestion
            notify_exception(None, f"Failed to get members of the IdP. {str(e)}")
```